internal/provider/resources/deployment_schedule.go

```diff
@@ -0,0 +1,382 @@
+			Timezone:   plan.Timezone.ValueString(),
+		},
+	}
+
+	err = client.Update(ctx, plan.DeploymentID.ValueUUID(), plan.ID.ValueUUID(), cfgUpdate)
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Deployment Schedule", "update", err))
+
+		return
+	}
+
+	schedules, err := client.Read(ctx, plan.DeploymentID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Deployment Schedule", "read after update", err))
+
+		return
+	}
+
+	resp.Diagnostics.Append(validateSchedules(schedules))
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	copyScheduleModelToResourceModel(schedules[0], &plan)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *DeploymentScheduleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state DeploymentScheduleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.DeploymentSchedule(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Deployment Schedule", err))
+
+		return
+	}
+
+	err = client.Delete(ctx, state.DeploymentID.ValueUUID(), state.ID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Deployment Schedule", "delete", err))
+
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func copyScheduleModelToResourceModel(schedule *api.DeploymentSchedule, model *DeploymentScheduleResourceModel) {
+	model.ID = customtypes.NewUUIDValue(schedule.ID)
+	model.Created = customtypes.NewTimestampPointerValue(schedule.Created)
+	model.Updated = customtypes.NewTimestampPointerValue(schedule.Updated)
+
+	model.DeploymentID = customtypes.NewUUIDValue(schedule.DeploymentID)
+
+	model.Active = types.BoolPointerValue(schedule.Active)
+	model.MaxActiveRuns = types.Float32Value(schedule.MaxActiveRuns)
+
+	model.Catchup = types.BoolValue(schedule.Catchup)
+	model.MaxScheduledRuns = types.Float32Value(schedule.MaxScheduledRuns)
+
+	model.Timezone = types.StringValue(schedule.Schedule.Timezone)
+	model.Interval = types.Float32Value(schedule.Schedule.Interval)
+	model.AnchorDate = types.StringValue(schedule.Schedule.AnchorDate)
+	model.Cron = types.StringValue(schedule.Schedule.Cron)
+	model.DayOr = types.BoolValue(schedule.Schedule.DayOr)
+	model.RRule = types.StringValue(schedule.Schedule.RRule)
+}
+
+//nolint:ireturn // required to return a diagnostic
+func validateSchedules(schedules []*api.DeploymentSchedule) diag.Diagnostic {
+	if len(schedules) != 1 {
+		return diag.NewErrorDiagnostic("Unsupported number of schedules", fmt.Sprintf("Expected 1 schedule, got %d. Only one schedule is supported.", len(schedules)))
+	}
+
+	return nil
+}
```

Comment:
We'll have to limit to 1 schedule at a time for now. Opened https://linear.app/prefect/issue/PLA-772/deployments-support-multiple-schedules to investigate support for multiple schedules.

Reply:
Added a note to the docs as well